### PR TITLE
separate data integrity logic from scraperController

### PR DESCRIPTION
**Pull request by yingjieMiao**

### CODE REVIEW

**Comment by yingjieMiao** [comment]
@leogrim 

initial attempt. still need more cleaning.

File: kifi-backend/modules/shoebox/app/com/keepit/admin/ScraperController.scala [hunk 1]
```diff
@@ -126,108 +120,5 @@ class ScraperController @Inject() (
     }
     Redirect(com.keepit.controllers.admin.routes.ScraperController.getUnscrapable())
   }
-
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/ScraperController.scala:129]: all these are moved to urlController now.

File: kifi-backend/modules/shoebox/app/com/keepit/admin/UrlController.scala [hunk 2]
```diff
@@ -177,4 +137,59 @@ class UrlController @Inject() (
...
+    }
+    Redirect(com.keepit.controllers.admin.routes.UrlController.documentIntegrity())
+  }
+
+  def documentIntegrity(page: Int = 0, size: Int = 50) = AdminHtmlAction { implicit request =>
+    val dupes = db.readOnly { implicit conn =>
+      duplicateDocumentRepo.getActive(page, size)
+    }
+
+    val groupedDupes = dupes.groupBy { case d => d.uri1Id }.toSeq.sortWith((a,b) => a._1.id < b._1.id)
```
Comment (gmethvin) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/UrlController.scala:156]:
``` scala
val groupedDupes = dupes.groupBy(_.uri1Id).toSeq.sortBy(_._1.id)
```

Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/UrlController.scala:156]: cool. This was copied from some old code. But I can change it.

File: kifi-backend/modules/shoebox/app/com/keepit/admin/UrlController.scala [hunk 3]
```diff
@@ -177,4 +137,59 @@ class UrlController @Inject() (
...
+
+    val loadedDupes = db.readOnly { implicit session =>
+      groupedDupes map  { d =>
```
Comment (gmethvin) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/UrlController.scala:159]: I usually like to destructure tuples with named variables to make things more readable:

``` scala
  groupedDupes map { case (uriId, dupes) =>
```

File: kifi-backend/modules/shoebox/app/com/keepit/admin/UrlController.scala (hunk 2 above)
Comment (gmethvin) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/UrlController.scala:156]: oh, totally missed the fact that it was copied.

**Pull request merged**